app/controllers/sms_controller.rb

```diff
@@ -75,9 +75,14 @@ def deliver_reply(reply)
       if incoming_adapter.reply_style == :via_adapter
         outgoing_adapter.deliver(reply)
       else # reply via response
-        incoming_adapter.prepare_message_for_delivery(reply)
+        begin
```

Comment:
Looks like this block is wrapping the wrong code. We had discussed earlier that replying via adapter is the part that might raise the error. So it should wrap the other half of this `if` block. You may need to adjust the test method call to use a different incoming adapter (e.g. Twilio) that has reply_style == `via_adapter`.

Reply:
left you a message on slack @smoyth